include/geometrycentral/surface/barycentric_vector.ipp

```diff
@@ -195,35 +195,58 @@ inline BarycentricVector BarycentricVector::rotated90(IntrinsicGeometryInterface
 
   switch (type) {
   case BarycentricVectorType::Face: {
-    double ui = faceCoords[0];
-    double uj = faceCoords[1];
-    double uk = faceCoords[2];
-    geom.requireEdgeLengths();
-    double l_ij = geom.edgeLengths[face.halfedge().edge()];
-    double l_jk = geom.edgeLengths[face.halfedge().next().edge()];
-    double l_ki = geom.edgeLengths[face.halfedge().next().next().edge()];
-    geom.unrequireEdgeLengths();
-    // coefficients of matrix D taking barycentric coords to 3D local coords, squared and multiplied by 2
-    double ai = l_ij * l_ij - l_jk * l_jk + l_ki * l_ki;
-    double aj = l_jk * l_jk - l_ki * l_ki + l_ij * l_ij;
-    double ak = l_ki * l_ki - l_ij * l_ij + l_jk * l_jk;
-    double s = 0.5 * (l_ij + l_jk + l_ki);
-    double A = std::sqrt(s * (s - l_ij) * (s - l_jk) * (s - l_ki));
-    Vector3 newCoords = {ak * uk - aj * uj, ai * ui - ak * uk, aj * uj - ai * ui};
-    newCoords /= (4. * A);
-    return BarycentricVector(face, newCoords);
+    return faceVectorRotated90(*this, geom);
     break;
   }
-  case BarycentricVectorType::Edge:
-    throw std::logic_error("Cannot rotate BarycentricVector of Type::Edge");
+  case BarycentricVectorType::Edge: {
+    // Convert to a face-type vector, then rotate.
+    Halfedge he = (edgeCoords[0] < 0.) ? edge.halfedge() : edge.halfedge().twin();
+    if (!he.isInterior()) {
+      he = he.twin();
+    }
+    BarycentricVector w = this->inFace(he.face());
+    return faceVectorRotated90(w, geom);
     break;
+  }
   default:
     // If vertex-type, do nothing
     break;
   }
   return *this;
 }
 
+inline BarycentricVector BarycentricVector::rotated(IntrinsicGeometryInterface& geom, double angle) const {
+
+  switch (type) {
+  case BarycentricVectorType::Face: {
+    return faceVectorRotated(*this, geom, angle);
+    break;
+  }
+  case BarycentricVectorType::Edge: {
```

Comment:
Do I read correctly that this takes in a `BarycentricVectorType::Edge`, but returns a `BarycentricVectorType::Face`? To me that would be confusing behavior, could we send it back to the edge before returning?

Reply:
(same for the other switch statement above)

Reply:
Ah this is well-documented though. I guess my question is _why_ do we want to return a vector on the adjacent face rather than keeping it on the edge?

Reply:
Good question -- I admit it feels weird. I guess there's a subtle technicality: This function is rotating a tangent vector, located in the tangent space of the edge, and this vector happens to be represented via barycentric coordinates on the edge. But edge-type vectors are constrained to point along the edge, since the space of edge-type vectors is only one-dimensional: there are two (dependent) barycentric coordinates, one corresponding to each vertex of the edge. So to do this rotation, we interpret the vector as living in its 2d tangent space, and re-interpret the rotated vector as a barycentric vector living in one of the adjacent faces. I'm open to ideas on how to make this more intuitive!